src/engine/GroupByHashMapOptimization.h

```diff
@@ -171,8 +185,10 @@ struct GroupConcatAggregationData {
   }
 
   // _____________________________________________________________________________
-  explicit GroupConcatAggregationData(std::string separator)
+  explicit GroupConcatAggregationData(std::string_view separator)
       : separator_{std::move(separator)} {
     currentValue_.reserve(20000);
   }
+
+  void reset() { currentValue_.clear(); }
```

Comment:
Maybe you could consider moving those `reset` functions also into the preparation PR, 
those seem pretty undisputed to me. Of course they should have tests then:)